{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "PQHE_notebook.ipynb",
      "provenance": [],
      "collapsed_sections": [],
      "authorship_tag": "ABX9TyPe/l+/OlyCHnbQ7LoNPvjQ",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "ir",
      "display_name": "R"
    },
    "language_info": {
      "name": "R"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/tgstoecker/cooperation/blob/main/PQHE_notebook.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ZVfRM93Zhu3n"
      },
      "source": [
        "# Pooled QTL Heritability Estimator (PQHE)\n",
        "[Estimation of QTL heritability based on pooled sequencing data](https://academic.oup.com/bioinformatics/article/34/6/978/4587583)  \n",
        "[https://github.com/biotangweiqi/PQHE](https://github.com/biotangweiqi/PQHE)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "68uxIUp_itqw"
      },
      "source": [
        "## 1. Installation of packages & cloning of the github repo\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "-VohLK8rhW4l"
      },
      "source": [
        "install.packages(\"rootSolve\")\n",
        "library(rootSolve)"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "w9Kx50EDi5nk"
      },
      "source": [
        "The github repository with the software will be installed at the root of the file system in a directory called 'PQHE'."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "cg7URUJxhyqk"
      },
      "source": [
        "setwd('/')\n",
        "system('git clone https://github.com/biotangweiqi/PQHE')"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "hcrhOZZUjda0"
      },
      "source": [
        "## 2. Creation of a conf. file with this layout:  \n",
        "(this is the layout of the input for our calculation)\n",
        "```\n",
        "b   = 0.25   # F2 population  \n",
        "m1  = 50     # individuals in pool-1  \n",
        "m2  = 50     # individuals in pool-2  \n",
        "n   = 531    # sample size of population  \n",
        "AF1 = 0.724  # estimated bulked allele frequency in pool-1  \n",
        "AF2 = 0.114  # estimated bulked allele frequency in pool-2  \n",
        "```"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "SDVG49kxnu9x"
      },
      "source": [
        "- **The following cell can be adapted to perform the analysis for each QTL.**\n",
        "- **Alternatively, have multiple cells - each for a different QTL.**\n",
        "- **This cell demonstratively peforms set up for the third QTL region (alternative allele comparison between mutant and WT bulk) - based on the  unfiltered_interesting_qtls_delta_snp.csv results file in our sciebo folder.**\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "7FRjtre_hy38"
      },
      "source": [
        "#moving into the PQHE directory\n",
        "setwd('/PQHE/')\n",
        "\n",
        "#we create an input file with our specifications\n",
        "fileConn<-file(\"input_exp1.txt\")\n",
        "writeLines(c\n",
        "            (\"b   = 0.25   # F2 population\",  \n",
        "             \"m1  = 143     # individuals in pool-1\",  \n",
        "             \"m2  = 143     # individuals in pool-2\",  \n",
        "             \"n   = 640    # sample size of population\",\n",
        "             \"AF1 = 0.964292979454104  # estimated bulked allele frequency in pool-1\",\n",
        "             \"AF2 = 0.364750850696872  # estimated bulked allele frequency in pool-2\"),\n",
        "            fileConn)\n",
        "close(fileConn)"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "0HnkxL8Fhy6F"
      },
      "source": [
        "#we run the programme with input we just created and write to an output object\n",
        "result <- system(\"Rscript estimate_QTL_heritability.v0.4.10.R input_exp1.txt\", intern=TRUE)\n",
        "#we print the results on a per-line basis \n",
        "cat(paste(result, collapse='\\n'))"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "gy8YyacFhy87"
      },
      "source": [
        ""
      ],
      "execution_count": null,
      "outputs": []
    }
  ]
}